clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -547,23 +549,35 @@ object GarbageCollector {
   private def getPreviousRunID(
       incrementalRun: Boolean,
       storageNS: String,
-      configMapper: ConfigMapper
+      configMapper: ConfigMapper,
+      iterations: Int
   ): String = {
     if (!incrementalRun) {
       return ""
     }
+    if (iterations < 1) {
+      throw RunIDException("Run ID iteration number cannot be smaller than 1")
+    }
     var previousRunID = ""
     val runIDsPath = new Path(s"${storageNS.stripSuffix("/")}/_lakefs/retention/gc/run_ids/")
     val runIDsFS = runIDsPath.getFileSystem(configMapper.configuration)
     val runIDs = runIDsFS.listFiles(runIDsPath, false)
-    if (runIDs.hasNext) {
-      previousRunID = runIDs.next().getPath.getName
-      println(s"----------------------- Using previous RUN ID: $previousRunID")
-    } else {
+    var remainingIterations = iterations
+    if (!runIDs.hasNext) {
       runIDsFS.close()
       throw RunIDException("No previous run ID")
     }
+    var runIDObject: LocatedFileStatus = null
+    while (runIDs.hasNext && remainingIterations > 0) {
+      remainingIterations -= 1
+      runIDObject = runIDs.next()
+    }
```

Comment:
This looks more like a _for_ loop situation.

Reply:
done